tests/test_models.py

```diff
@@ -213,9 +212,216 @@ def test_wishlist_id_is_increment(self):
+        self.assertEqual(item.wishlist_id, values["wishlist_id"])
+        self.assertEqual(item.product_id, values["product_id"])
+        self.assertEqual(item.product_name, values["product_name"])
+        self.assertEqual(item.product_price, values["product_price"])
+        self.assertEqual(item.created_date, values["created_date"])
+
+    def test_wishlist_item_repr_method(self):
+        """It should give a correct response when calling __repr__"""
+        item = WishlistItemFactory()
+        expected = f"WishlistItem(id={item.id}, wishlist_id={item.wishlist_id}, product_id={item.product_id}, product_name='{item.product_name}', product_price={item.product_price}, quantity={item.quantity}, created_date='{item.created_date}')"
+        self.assertEqual(repr(item), expected)
+
+    def test_wishlist_item_serialize(self):
+        """It should serialize a wishlist item"""
+        item = WishlistItemFactory()
+        serialized = item.serialize()
+        self.assertEqual(item.wishlist_id, serialized["wishlist_id"])
+        self.assertEqual(item.product_id, serialized["product_id"])
+        self.assertEqual(item.product_name, serialized["product_name"])
+        self.assertEqual(item.product_price, serialized["product_price"])
+        self.assertEqual(item.created_date, serialized["created_date"])
+
+    def test_wishlist_item_deserialize(self):
+        """It should deserialize a wishlist"""
+
+        item = WishlistItem()
+        item.deserialize(self.serialized)
+        self.assertEqual(item.id, self.serialized["id"])
+        self.assertEqual(item.wishlist_id, self.serialized["wishlist_id"])
+        self.assertEqual(item.product_id, self.serialized["product_id"])
+        self.assertEqual(item.product_name, self.serialized["product_name"])
+        self.assertEqual(item.product_price, self.serialized["product_price"])
+        self.assertEqual(item.quantity, self.serialized["quantity"])
+        self.assertEqual(item.created_date, self.serialized["created_date"])
+
+    def test_wishlist_item_deserialize_with_invalid_data_type(self):
+        """It should raise a DataValidationError when serializing from incorrect type"""
+
+        item = WishlistItem()
+        self.assertRaises(
+            DataValidationError,
+            item.deserialize,
+            "You can't deserialize me. I'm a mighty STRING!!!",
+        )
+
+    def test_wishlist_item_deserialize_with_invalid_input(self):
+        """It should throw a DataValidationError when a serialized object is missing a key"""
+
+        del self.serialized["product_name"]
+        item = WishlistItem()
+        self.assertRaises(DataValidationError, item.deserialize, self.serialized)
+
+    # don't love this approach.  This is why it feels slightly wrong to me to unit test with DB
+    # I have to rely on the wishlist Model functioning for this test to pass otherwise the Foreign Key
+    # constraint will fail when trying to save an arbitrary wishlist_id
```

Comment:
I get it, but at least we have tests for the wishlist model to pass at least in case it does fail. 
Otherwise, it looks good.